Explorer/Assets/DCL/Audio/WorldAudioPlaybackController.cs

```diff
@@ -0,0 +1,155 @@
+using Random = UnityEngine.Random;
+
+namespace DCL.Audio
+{
+    public class WorldAudioPlaybackController : MonoBehaviour, IDisposable
+    {
+        [SerializeField]
+        private WorldAudioSettings audioSettings;
+        [SerializeField]
+        private AudioSource audioSourcePrefab;
+
+        private readonly Dictionary<int, List<AudioSource>> landscapeAudioSourcesPerIndex = new ();
+        private readonly Dictionary<int, List<AudioSource>> oceanAudioSourcesPerIndex = new ();
+        private GameObjectPool<AudioSource> audioSourcePool;
+
+        public void Dispose()
+        {
+            WorldAudioEventsBus.Instance.PlayLandscapeAudioEvent -= OnPlayLandscapeAudioEvent;
+            WorldAudioEventsBus.Instance.StopWorldAudioEvent -= StopAndReleaseAudioSources;
+            audioSourcePool.Dispose();
+        }
+
+        public void Initialize()
+        {
+            WorldAudioEventsBus.Instance.PlayLandscapeAudioEvent += OnPlayLandscapeAudioEvent;
+            WorldAudioEventsBus.Instance.StopWorldAudioEvent += StopAndReleaseAudioSources;
+            audioSourcePool = new GameObjectPool<AudioSource>(transform, OnCreateAudioSource);
+        }
+
+        private AudioSource OnCreateAudioSource()
+        {
+            AudioSource audioSource = Instantiate(audioSourcePrefab, transform);
+            audioSource.outputAudioMixerGroup = audioSettings.MixerGroup;
+            audioSource.loop = true;
+            audioSource.spatialBlend = 1;
+            return audioSource;
+        }
+
+
+        private AudioSource GetAudioSource(int2 position)
+        {
+            AudioSource audioSource = audioSourcePool.Get();
+            Transform audioSourceTransform = audioSource.transform;
+            audioSourceTransform.parent = transform;
+            audioSourceTransform.position = new Vector3(position.x, 1, position.y);
+            return audioSource;
+        }
+
+        private void SetupAudioClip(AudioSource audioSource, AudioClipConfig audioClipConfig)
+        {
+            int clipIndex = AudioPlaybackUtilities.GetClipIndex(audioClipConfig);
+            var clip = audioClipConfig.AudioClips[clipIndex];
+            audioSource.clip = clip;
+            audioSource.time = Random.Range(0, clip.length);
+            audioSource.volume = audioClipConfig.RelativeVolume;
+            audioSource.Play();
+        }
+
+
+
+        private void OnPlayLandscapeAudioEvent(int index, NativeArray<int2> audioSourcesPositions, WorldAudioClipType clipType)
+        {
+            AudioClipConfig audioClipConfig = audioSettings.GetAudioClipConfigForType(WorldAudioClipType.Landscape).DayClip; //We can switch this depending on the time of day for example
+
+            if (!CheckAudioClips(audioClipConfig)) return;
+
+            var audioSourcesPerIndex = landscapeAudioSourcesPerIndex;
+
+            switch (clipType)
+            {
+                default:
+                case WorldAudioClipType.Landscape:
```

Comment:
There is a bit of magic happening here.
It feels like it should be connected with `WorldAudioClipType` from `WorldAudioSettings` but it's not clear how. Can you write some comments for clarification?

Reply:
I straightened it up a bit and created a new dictionary that holds the other dictionaries, its a bit more clean I think now